### PR TITLE
Refactor Pitch Doc from cards to table

### PR DESCRIPTION
**Pull request by ayanmallik**
## Summary

Utilize Bryan's [redesigns](https://www.figma.com/file/MHRVK2dRznJ3mSogT57dM5/%5BFa21%5D-SSW-Bryan's-Designs?node-id=598%3A45494) to refactor pitches from cards to table

## Changes

- Modularize Directory Table to generalize the table header/body components
- Extract interests and teams from pitches into the TableTool component
- Implement Sahi/Bryan's logic to determine whether teams show up as claimable on table row

## Screenshots

image

### CODE REVIEW

File: client/src/components/Tables/PitchDoc/index.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  pitches,
+  sortedPitches,
+  setSortedPitches,
+}): ReactElement => {
+  const [column, setColumn] = useState<ColumnEnumValue>();
+  const [direction, setDirection] = useState<'ascending' | 'descending'>();
+
+  const handleSort = (newColumn: ColumnEnumValue): void => {
+    if (column?.title === newColumn.title) {
+      if (direction === 'ascending') {
```
Comment (nkonjeti) [inline client/src/components/Tables/PitchDoc/index.tsx:60]: you could use a switch case statement but this also looks good

**Review by nkonjeti — APPROVED** [review]
Hi Ayan everything looks good! Just added one comment:)

File: client/src/components/Tables/PitchDoc/index.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        </Table.Cell>
+      </Table.Row>
+    )}
+    {sortedPitches.map((user, index) => (
+      <PitchRow pitch={user} key={index} />
+    ))}
+  </>
+);
+
+const getClaimableTeams = (pitch: IPitch, user: IUser): string[] =>
```
Comment (amit-sawhney) [inline client/src/components/Tables/PitchDoc/index.tsx:120]: aye nice. @sahithimuthyala double check this.

File: client/src/components/Tables/PitchDoc/index.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  return (
+    <Table.Row>
+      <Table.HeaderCell
+        onClick={() => handleSort(columnsEnum.TITLE)}
+        sorted={column === columnsEnum.TITLE ? direction : undefined}
+      >
+        Title
+      </Table.HeaderCell>
+      <Table.HeaderCell
+        onClick={() => handleSort(columnsEnum.DESCRIPTION)}
```
Comment (amit-sawhney) [inline client/src/components/Tables/PitchDoc/index.tsx:94]: I don't think we need to be able to sort by description. @bryange3 @sahithimuthyala double check me on that

**Review by amit-sawhney — APPROVED** [review]
This looks great Ayan! 💯 Way to generalize the tables, that will make development going forward insanely easy. One small thing,

image

It appears that this one tab on the Admin View for Pitches pending approval has a different table. If you want, I can assign this to you in another issue or you can fix it on this PR. Let me know which you would prefer.

@bryange3 can you confirm that this is the only table exception to the Pitch Doc

Also, could you fix the issue that this closes (not 85 and 22)

**Pull request merged**